src/libraries/Common/src/Roslyn/Hash.cs

```diff
@@ -0,0 +1,381 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+
+namespace Roslyn.Utilities
+{
+    internal static class Hash
+    {
+        /// <summary>
+        /// This is how VB Anonymous Types combine hash values for fields.
+        /// </summary>
+        internal static int Combine(int newKey, int currentKey)
+        {
+            return unchecked((currentKey * (int)0xA5555529) + newKey);
+        }
+
+#if false
```

Comment:
if i was only referencing a tiny part of the surface area of the copied code, i `ifdefed` out the rest to keep teh intrusion minimal.  i kept the code in case we may need to access more of it.  this allows the code to stay close in sync with teh roslyn version.